float64[] path_posx
float64[] path_posy
uint8 pathFlag
